diff -Nuar libcxx.orig/include/cstddef libcxx/include/cstddef
--- libcxx.orig/include/cstddef	2015-10-09 00:26:50.000000000 +0000
+++ libcxx/include/cstddef	2017-04-23 17:10:01.944023536 +0000
@@ -48,7 +48,8 @@
 using ::ptrdiff_t;
 using ::size_t;
 
-#if defined(__CLANG_MAX_ALIGN_T_DEFINED) || defined(_GCC_MAX_ALIGN_T)
+#if defined(__CLANG_MAX_ALIGN_T_DEFINED) || defined(_GCC_MAX_ALIGN_T) || \
+    defined(__DEFINED_max_align_t)
 // Re-use the compiler's <stddef.h> max_align_t where possible.
 using ::max_align_t;
 #else
diff -Nuar libcxx.orig/include/stddef.h libcxx/include/stddef.h
--- libcxx.orig/include/stddef.h	2015-10-09 00:26:50.000000000 +0000
+++ libcxx/include/stddef.h	2017-04-23 17:10:01.944023536 +0000
@@ -53,7 +53,8 @@
 }
 
 // Re-use the compiler's <stddef.h> max_align_t where possible.
-#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T)
+#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T) && \
+    !defined(__DEFINED_max_align_t)
 typedef long double max_align_t;
 #endif
 
diff: libcxx.orig/test/std/experimental/filesystem/Inputs/static_test_env/bad_symlink: No such file or directory
diff: libcxx/test/std/experimental/filesystem/Inputs/static_test_env/bad_symlink: No such file or directory
